modules/internal/ChapelIO.chpl

```diff
@@ -465,7 +465,7 @@ module ChapelIO {
         if st == QIO_AGGREGATE_FORMAT_JSON {
           start = new ioLiteral("{");
         } else if st == QIO_AGGREGATE_FORMAT_CHPL {
-          start = new ioLiteral("new " + typeToString(t) + "(");
+          start = new ioLiteral("new " + typeToString(t).c_str() + "(");
```

Comment:
Right, I think you understood me. I think there is some philosophical distinction here that is not obvious to me:
  * .c_str() returns a pointer to the string's internal data
  * : c_string does not ? it returns a full copy? couldn't it return a pointer to the internal data?

I think it is probably a better idea to disable all : c_string casts (although param ones might be a useful exception) since seeing c_str() should remind a reader of some code that it's creating a temporary pointer inside the string... and I think that property would be less obvious with a cast.

So, all that is to say, I don't think we need to change how cast to c_string or .c_str() work right now.